### ISSUE TITLE
Forcing SSL

### ISSUE THREAD

**Issue by kalmanh** [simplabs/rails_api_auth#29]
Does it ever make sense NOT to use SSL when requesting a token? Should SSL be forced when requesting/revoking a token?

(It might make sense to be more flexible inside the "authenticated" controller - for example read operations probably don't need SSL, but write ones do... but inside the "token" controller - I am thinking it should always be SSL forced down your throat).

http://api.rubyonrails.org/classes/ActionController/ForceSSL/ClassMethods.html

Thoughts?

**Comment by marcoow** [comment]
Yeah, actually you should have SSL enabled at all times when requesting a token (and also just always and for everything actually). I'm not sure whether we should solve this by adding code that does it or by simply adding documentation that reminds devs of switching on SSL when deploying this to production.

You might also be running this only internally so that it only receives requests from other APIs that are accessible from the internets then. In that case you could argue SSL isn't really required.

**Comment by kalmanh** [comment]
@marcoow - I believe there are 2 distinct scenarios:

Scenario 1: _(This is the scenario you are describing)_ Your website always needs to be authenticated. You can't DO ANYTHING without putting in your username and password. This scenario is simple to implement. Inside `production.rb`

```
config.force_ssl = true
```

Everyone is happy - no changed to `rails_api_auth` gem needed.

Scenario 2: Your website traffic is READ-heavy. For a lot of the pages, you don't need or want authentication and you don't want SSL. But for say an admin section of your website, you DO want SSL. In this case, you want to deploy your website WITHOUT SSL and turn on SSL (AND FORCE) for individual controllers (including the OAuth controller) using something like [force_ssl](http://api.rubyonrails.org/classes/ActionController/ForceSSL/ClassMethods.html)

I am currently implementing a Scenario 2 type application and I would like to constrain the Oauth 2 controller to ALWAYS force SSL. 

So, what I am proposing is as follows (and I can PR this if you agree that this is a good idea):

1. Have force_ssl inside [Oauth2Controller](https://github.com/simplabs/rails_api_auth/blob/master/app/controllers/oauth2_controller.rb#L7) with an [if](https://github.com/rails/rails/blob/4a2c7c6e7ec88759ae5ac2489b093a2b8330809c/actionpack/lib/action_controller/metal/force_ssl.rb#L32) option

2. Add an accessor to [RailsApiAuth](https://github.com/simplabs/rails_api_auth-demo/blob/master/config/initializers/rails_api_auth.rb#L1) module that can be added through configuration in the initializer, something like `lockdown_with_ssl`. Something like:

```
raa.lockdown_with_ssl = true
```

The default for this value could be `false`, so nothing will really change for existing apps or even future apps, UNLESS you configure this new value to be `true`

What do you think?

**Comment by marcoow** [comment]
Yeah, sounds like adding 

```rb
RailsApiAuth.force_ssl
```

which would default to `true` would probably be a good idea.

**Comment by kalmanh** [comment]
You want me to PR this one?

**Comment by marcoow** [comment]
Sure, would be great

**Comment by marcoow** [comment]
implemented in #34
